@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -0,0 +1,1268 @@
+                for (
+                    ;
+                    groupIndex < groupCount && groups[groupIndex].technique === techniqueIndex;
+                    ++groupIndex
+                ) {
+                    if (start + count !== groups[groupIndex].start) {
+                        break;
+                    }
+
+                    count += groups[groupIndex].count;
+                }
+
+                const ObjectCtor = getObjectConstructor(technique);
+
+                if (ObjectCtor === undefined) {
+                    continue;
+                }
+
+                let material: THREE.Material | undefined = materials[techniqueIndex];
+
+                if (material === undefined) {
+                    const onMaterialUpdated = (texture: THREE.Texture) => {
+                        dataSource.requestUpdate();
+                        if (texture !== undefined) {
+                            tile.addOwnedTexture(texture);
+                        }
+                    };
+                    material = createMaterial(
+                        {
+                            technique,
+                            level: Math.floor(zoomLevel),
+                            fog: mapView.scene.fog !== null
+                        },
+                        onMaterialUpdated
+                    );
+                    if (material === undefined) {
+                        continue;
+                    }
+                    materials[techniqueIndex] = material;
+                }
+
+                // Modify the standard textured shader to support height-based coloring.
+                if (
+                    isTerrainTechnique(technique) &&
+                    technique.heightBasedColors !== undefined &&
+                    technique.displacementMap !== undefined
+                ) {
+                    (material as any).onBeforeCompile = (shader: THREE.Shader) => {
+                        shader.fragmentShader = shader.fragmentShader.replace(
+                            "#include <map_pars_fragment>",
+                            `#include <map_pars_fragment>
+    uniform sampler2D displacementMap;
+    uniform float displacementScale;
+    uniform float displacementBias;`
+                        );
+
+                        shader.fragmentShader = shader.fragmentShader.replace(
+                            "#include <map_fragment>",
+                            `#ifdef USE_MAP
+    float minElevation = ${EarthConstants.MIN_ELEVATION.toFixed(1)};
+    float maxElevation = ${EarthConstants.MAX_ELEVATION.toFixed(1)};
+    float elevationRange = maxElevation - minElevation;
+
+    float disp = texture2D( displacementMap, vUv ).x * displacementScale + displacementBias;
+    vec4 texelColor = texture2D( map, vec2((disp - minElevation) / elevationRange, 0.0) );
+    texelColor = mapTexelToLinear( texelColor );
+    diffuseColor *= texelColor;
+#endif`
+                        );
```

Comment:
this is missing the latest change I did, compare Tile.ts on master.

Reply:
Will merge, thanks!